### PR TITLE
Phase 0 UX overhaul: theme + dark mode, dialogs, validation, empty states, polish, green rebrand (roadmap 0.6–0.10)

### PR DESCRIPTION
**Pull request by bberardi**
## Summary

The complete **Phase 0 UX overhaul** (roadmap items 0.6–0.10) plus a green brand refresh — one commit per item, each landed green through the full check suite. Test suite grew 202 → **235 passing**; lint, format, and build green at every step.

| Commit | Item | What it does |
| --- | --- | --- |
| `5739b31` | **0.6 Theme system** | Proper MUI 9 theme in `src/theme/` codifying the existing look (palette, typography, component defaults replacing repeated ad-hoc `sx`); **persisted dark mode** via the color-schemes API with `InitColorSchemeScript` (no first-paint flash); centralized `formatCurrency`/`formatPercent` in a pure `src/helpers/format-helpers.ts` replacing six inline duplicates with byte-for-byte identical output (14 parity-guarding tests) |
| `45e22e1` | **0.7 Dialogs** | All six Popover-as-modal surfaces converted to a shared `ResponsiveDialog` (full-screen on mobile, proper focus trap/Escape); delete moved out of the edit form to the table-row actions behind a confirmation dialog; **snackbar soft-undo** restores the deleted entity at its original index via new pure reducer actions (9 tests incl. list-shrank and double-undo edge cases) |
| `af62a88` | **0.8 Validation UX** | Field-level error messages with touched/submit-attempt tracking (empty add form doesn't open all-red) and an always-visible explanation of why Save is disabled; **non-blocking cross-field sanity warnings** (CurrentAmount > Principal, rates > 30%/20% thresholds, contribution cadence finer than compounding). All rules in one pure `validation-helpers.ts` (24 tests) that `isFormValid()` derives from — save semantics unchanged |
| `dbfb636` | **0.9 Empty states & sample data** | "Welcome to PathWise" onboarding state with Add/Load-sample CTAs; lighter per-section empty states; realistic `Sample:`-prefixed entities (mortgage, car loan, index fund, 401(k)) in `src/state/sample-data.ts`; dismissible "Showing sample data" banner with one-click clear; **Test Data dev switch removed** from the command bar. Reducer stash/restore guarantees preserved under renamed `LoadSampleData`/`ClearSampleData` actions |
| `6bd27df` | **0.10 Layout polish** | Command-bar hierarchy (contained Add buttons as primary; tooltipped import/export as secondary); toolbar wraps cleanly at ~360px via theme override; repeated px spacing replaced with theme tokens; footer gains GitHub link + app version injected from package.json via Vite `define` |
| `de531c4` | **Green rebrand** | Brand identity moved from blue to a finance green palette, contained entirely in `src/theme/theme.ts`: emerald primary (`#00875a` light / `#00e676` dark), deep forest gradient (`#0b3d2e → #14532d`) on the AppBar pill/Header/Footer, green-tinted dark backgrounds (`#0c1512`). Error/warning semantics left standard so validation and delete affordances still read as alerts |

## Notes for review

- **Known follow-up**: the logo PNGs in `src/assets/` are still the navy/cyan originals — they sit on a white card so nothing breaks, but the mark reads off-brand against the green identity. Needs the asset re-exported in the green palette (not addressable in code).
- Dark mode is worth a look in the deployed preview: green-cast near-black surfaces with bright `#00e676` accents, per the "dark but bright green" direction.
- No new dependencies anywhere in the stack; version stays 0.7.0 (the Phase 0 target).
- With this merged, Phase 0 is done except **0.11 (Math Verification Suite)** — the Charter implementation, planned as its own 2–3 PR sequence since it adds dev tooling (fast-check, Stryker, coverage gates).

https://claude.ai/code/session_01J8KWM4siLf8tAZL5ot2qqn

---
_Generated by [Claude Code](https://claude.ai/code/session_01J8KWM4siLf8tAZL5ot2qqn)_

### CODE REVIEW

File: src/loan/loan-table.tsx [hunk 1]
```diff
@@ -85,6 +69,14 @@ export const LoanTable = (props: LoanTableProps) => {
...
+      <IconButton
+        onClick={() => props.onLoanDelete(loan)}
+        color="error"
+        size={isMobile ? 'medium' : 'small'}
+        title="Delete Loan"
+      >
+        <Delete />
+      </IconButton>
     </Box>
   );
```
Comment (bberardi) [inline src/loan/loan-table.tsx]: `LoanActions` (and `LoanCard` below it) are component functions defined inside `LoanTable`'s render. Because their identity is re-created on every parent render, React treats them as a brand-new component type each time and unmounts/remounts the entire subtree per row on every re-render of the table (e.g. when `selectedPit`/`selectedAmortization` change). That throws away DOM/state for every row each time the parent updates.

`InvestmentActions`/`InvestmentCard` in `investment-table.tsx` have the same issue, and `LoanActions`/`InvestmentActions` are near-identical (same Box layout, same four icon buttons, only the handlers and labels differ).

Hoist these to module-level components that take their callbacks as props — that fixes the remount churn and lets the two tables share one `EntityRowActions`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/loan/add-edit-loan.tsx [hunk 2]
```diff
@@ -1,58 +1,89 @@
...
+  // touched fields (or attempted save) we list the specific revealed errors;
+  // before that we show a neutral prompt so the form doesn't open all-red but
+  // the user still knows what's expected.
+  const revealedErrors = (Object.keys(validation.errors) as LoanField[])
+    .filter((field) => showFor(field))
+    .map((field) => validation.errors[field]);
+  const saveDisabledReason =
+    revealedErrors.length > 0
+      ? revealedErrors.join(' ')
+      : 'Fill in all required fields to enable saving.';
```
Comment (bberardi) [inline src/loan/add-edit-loan.tsx]: This whole field-tracking block — `touched`/`submitAttempted` state, `showFor`, `touch`, `errorFor`, `warningFor`, `resetTracking`, `revealedErrors`, and `saveDisabledReason` (lines 33–64) — is byte-for-byte identical to `add-edit-investment.tsx` (lines 38–74), only the field-type parameter differs. This PR deliberately centralized formatting and validation into shared pure helpers; this scaffolding is the one piece that stayed duplicated, so the two forms can now silently drift (e.g. a change to how the disabled-reason is composed has to be made in two places).

Consider extracting a generic `useFieldTracking<TField>(validation)` hook that returns `{ showFor, touch, errorFor, warningFor, resetTracking, revealedErrors, saveDisabledReason }`, matching the PR's own pattern.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/loan/add-edit-loan.tsx [hunk 3]
```diff
@@ -79,227 +110,254 @@ export const AddEditLoan = (props: AddEditLoanProps) => {
...
+                      ? newValue[0]
+                      : newValue,
                   })
                 }
-                views={['year', 'month', 'day']}
-                openTo="year"
+                valueLabelDisplay="auto"
+                step={0.25}
+                min={0}
+                max={30}
```
Comment (bberardi) [inline src/loan/add-edit-loan.tsx]: `max={30}` is the literal value of `LOAN_RATE_WARNING_THRESHOLD` (validation-helpers.ts:39), but hardcoded here rather than imported. Two consequences:

1. The slider can never reach a rate that triggers the "Interest rate above 30%" warning — that warning fires only on `> 30`, so it's reachable solely through the manual text field. The slider's ceiling and the warning boundary are coincidentally equal, which makes the warning effectively dead for slider users.
2. If the threshold is ever changed in validation-helpers, the slider won't follow and the two will diverge.

Importing `LOAN_RATE_WARNING_THRESHOLD` and setting e.g. `max={LOAN_RATE_WARNING_THRESHOLD + 5}` (or some deliberate ceiling) keeps the input range and the warning rule tied to one source. Minor, but worth a look since the PR otherwise made a point of single-sourcing these rules.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/state/finance-data-context.tsx [hunk 4]
```diff
@@ -52,11 +58,13 @@ export const FinanceDataProvider = ({ children }: { children: ReactNode }) => {
...
       importMerge: (loans: Loan[], investments: Investment[]) =>
         dispatch({ type: 'ImportMerge', loans, investments }),
-      enableTestData: (loans: Loan[], investments: Investment[]) =>
-        dispatch({ type: 'EnableTestData', loans, investments }),
-      disableTestData: () => dispatch({ type: 'DisableTestData' }),
+      loadSampleData: (loans: Loan[], investments: Investment[]) =>
+        dispatch({ type: 'LoadSampleData', loans, investments }),
+      clearSampleData: () => dispatch({ type: 'ClearSampleData' }),
     }),
     [state]
```
Comment (bberardi) [inline src/state/finance-data-context.tsx]: The `useMemo` depends on `[state]`, so all nine action creators (`addLoan`, `deleteLoan`, `insertLoanAt`, …) are reconstructed on every state change even though each one closes over only the stable `dispatch` and its own arguments — none of them actually reads `state`. Any consumer that destructures just a method off the context still sees a new function identity on every add/update/delete, defeating downstream memoization.

You could define the action creators once (module-level factory over `dispatch`, or `useCallback` with `[]`) and only re-memo the `{ state, ...creators }` object when `state` changes. Minor, but it's cheap to get the stable identities.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by bberardi — COMMENTED** [review]
Reviewed the Phase 0 UX overhaul. This is a clean, well-tested PR — the validation/formatting centralization and the stash/restore reducer logic are solid, and the soft-undo restore-at-index handling (with clamping and the already-present guard) is correct. I left four inline comments, all non-blocking; the most worthwhile is the nested-component-in-render pattern in the tables (causes per-render remounts) and the duplicated form field-tracking scaffolding that could become a shared hook. Nothing here blocks merge.

A few candidates I checked and cleared, for the record:
- The `revealedErrors.join(' ')` in both forms can't emit a literal "undefined" — `Object.keys(validation.errors)` only yields keys that were assigned string values.
- The contribution-cadence warning can't misfire: `ContributionFrequency` is typed as `CompoundingFrequency`, so `cadenceRank` never returns -1.
- `ConfirmDeleteDialog`'s `fullScreen={false}` is intentional and documented, not a missed responsive case.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/loan/loan-table.tsx (hunk 1 above)
Comment (bberardi) [reply src/loan/loan-table.tsx]: Fixed in b8a5d81. Hoisted the row/card components to module scope and extracted a shared `EntityRowActions` (`src/components/entity-row-actions.tsx`) that both tables feed an `actions` list — no more per-render component redefinition (so rows no longer remount when a popout opens), and the duplicated button cluster is gone.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/loan/add-edit-loan.tsx (hunk 2 above)
Comment (bberardi) [reply src/loan/add-edit-loan.tsx]: Fixed in b8a5d81. Extracted a generic `useFieldTracking<TField>(validation)` hook (`src/helpers/use-field-tracking.ts`); both forms now consume it, so the touched/submit-attempt reveal logic and the save-disabled message live in one place.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/loan/add-edit-loan.tsx (hunk 3 above)
Comment (bberardi) [reply src/loan/add-edit-loan.tsx]: Fixed in b8a5d81. The slider now imports `LOAN_RATE_WARNING_THRESHOLD` and sets `max={LOAN_RATE_WARNING_THRESHOLD + 10}`, so the warning is reachable via the slider and the ceiling tracks the threshold instead of drifting.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/state/finance-data-context.tsx (hunk 4 above)
Comment (bberardi) [reply src/state/finance-data-context.tsx]: Fixed in b8a5d81. The action creators are now built once in a `useMemo([])` (they only close over the stable `dispatch`), and the `value` object re-memoizes on `[state, actions]` — so method identities stay stable across unrelated state changes.

---
_Generated by [Claude Code](https://claude.ai/code)_
